### ISSUE TITLE
Fix accessibility problems in the mobile navigation menu 

### ISSUE THREAD

**Issue by roslynwythe** [hackforla/website#4301]
### Overview

As developers we need to add the appropriate ARIA tags to the mobile "hamburger" icon/button and the mobile popup menu in order to enable use by AT (Assistive Technologies) such as screen readers.  In this item we will provide functional and status information to the AT so that mobile device users can navigate the HfLA website using gestures or an external keyboard.  

### Details
Currently, mobile device users cannot use AT to navigate the HfLA website, because the menu button and menu items lack insufficient WAI-ARIA attributes.  In particular, 
  - The hamburger icon/button used to expand the menu lacks a descriptive role/name and so it is not announced appropriately to the user
  - There is no `<nav>` element present in the mobile view until the hamburger icon/button is activated
  - The menu lacks the `aria-expanded` attribute to inform the AT whether the menu is expanded or hidden
  - Once the menu is expanded, the menu items are not navigable via keyboard or gestures, because they are not in the default tab order and they lack the `taborder` attribute

### Action Items
- [x] Referring to the [W3C ARIA Authoring Practices Guide - Navigation Menu Button Example](https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-links/), make the role and name of the hamburger icon/button explicit using these WAI-ARIA attributes: type, aria-haspopup, aria-controls, aria-name  
- [x] Use JavaScript to dynamically assign the `aria-expanded` attribute to indicate when the menu is in the hidden vs the expanded state.  Alternatively, create two separate `<button>` elements, one for displaying the menu and the other for collapsing the menu, and mark each with an appropriate WAI-ARIA label.  
- [x] Assign `role` and `aria-labelledby` attributes of the `<ul>` element
- [x] Assign `role="none"` attribute of the `<li>` element
- [x] Assign the `role="menuitem"` of the `<a>` element
- [x] In addition, to fulfill the requirement for a text description of the hamburger SVG icon:
   - [x] Add the following attribute to the `<svg>` element: aria-labelledby (reference the button)
- [x] Test the code locally using a screen reader and ensure that the screen reader announces the purpose of the hamburger icon/button, and when the button is activated, the user can use the Tab keys to navigate between menu items and the Enter or Space key to select a menu link.  When the menu is displayed, the screen reader should indicate that. 

### Resources/Instructions
 From the IDE, HTML for the header can be found in `_includes/header.html` which in turn includes `_includes/hamburger-nav.html` which renders the hamburger icon/button and list of links that comprise the mobile navigation menu.  The JavaScript to generate the links can be found in `assets/js/hamburger-nav.js`.
- [Google Chrome Screen Reader Extension (Chromevox)](https://chrome.google.com/webstore/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn?hl=en) is a free option for screen reader testing
  - Shortcuts
    -  use Tab to move to the next selectable item and Shift +Tab to move to the previous selectable item
    - use Enter to perform the default action on the current item. 
- [Hamburger Nav HTML](https://github.com/hackforla/website/blob/gh-pages/_includes/hamburger-nav.html)
- [JavaScript for Mobile Navigation](https://github.com/hackforla/website/blob/gh-pages/assets/js/hamburger-nav.js)
- [W3C ARIA Authoring Practices Guide - Navigation Menu Button Example](https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-links/)
- [MDN - aria-expanded](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded)
- [Using the WAI-ARIA expanded state to mark expandable and collapsible regions](https://www.w3.org/WAI/GL/wiki/Using_the_WAI-ARIA_aria-expanded_state_to_mark_expandable_and_collapsible_regions)
- [Android Accessibility - Use Talk Back to Browse the web](https://support.google.com/accessibility/android/answer/2633135?sjid=2647301033911588737-NA#explore_reading_controls)
- This item was created as a result of #4300

**Comment by roslynwythe** [comment]
@JessicaLucindaCheng In this issue I specified the addition of an aria-label to the `<button>` element which encloses the hamburger navigation icon svg.  This technique isn't explicitly mentioned in your wiki but I believe it is equivalent to addition of the aria-label to an enclosing `div`.  The addition of a `role` attribute isn't required because the `<button>` element is equivalent to a role.  Does this sound ok?

**Comment by JessicaLucindaCheng** [comment]
@roslynwythe I didn't think about button elements with SVGs when I wrote the wiki because I was focusing on the SVGs we use purely as images. So, I lack knowledge of how to apply buttons with ARIA labels. Would you be willing to do a little research to figure out the best way to apply aria-labels to SVGs and other images we use in buttons? If yes, we can put that information in an ER or a new issue and confer with Vanessa on where the best place to put it is when the website-wiki repo is ready for contributions. Let me know.

Note: Here is a MDN Web Docs article: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role

**Comment by roslynwythe** [comment]
> @roslynwythe I didn't think about button elements with SVGs when I wrote the wiki because I was focusing on the SVGs we use purely as images. So, I lack knowledge of how to apply buttons with ARIA labels. Would you be willing to do a little research to figure out the best way to apply aria-labels to SVGs and other images we use in buttons? If yes, we can put that information in an ER or a new issue and confer with Vanessa on where the best place to put it is when the website-wiki repo is ready for contributions. Let me know.
> 
> Note: Here is a MDN Web Docs article: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role

@JessicaLucindaCheng Yes, absolutely, I'll start with the research.  I had been thinking of adding to your recent wiki on accesible SVGs, but perhaps buttons do merit their own page of documentation.  Thanks!

**Comment by roslynwythe** [comment]
@JessicaLucindaCheng You were right, this issue did need more research!  Testing with a screen reader revealed a serious accessibility problem with the navigation menu, and raised a few other questions.  Fortunately ARIA provides some special attributes to handle navigation menu buttons and menu items, with a sample provided [here](https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-links/)  .  This issue will require overhaul and additional issues may be required to address all the problems, so I created ER #4300.  Should it be discussed at a meeting?

**Comment by JessicaLucindaCheng** [comment]
@roslynwythe If you want to discuss it with the dev team and/or dev leads at a meeting for input/feedback, you can do that. Usually, I like to run a plan by Bonnie at a Dev/PM meeting to let her know what's going on, get her feedback/input on it, and get approval before enacting the plan.

**Comment by ExperimentsInHonesty** [comment]
@roslynwythe I'm ready to prioritize this, if you are.  If you are ready to prioritize it, go ahead and add the `ready for prioritization` label and ill catch it tomorrow.

**Comment by ExperimentsInHonesty** [comment]
- @roslynwythe On this issue, there are quite a few notes about updating the wiki to include navigation items accessibility (above), that I think could be moved to an ER (if there isn't one already). There is also reference to an ER #4300, but it's actually an Epic. I will prioritize this issue after the comments are moved to the ER and hidden.

**Comment by roslynwythe** [comment]
@ExperimentsInHonesty #4300 has been updated to include the wiki issue, and comments in this issue have been hidden.

**Comment by allison-truhlar** [comment]
Availability - through Nov 21, I'm available 9-5pm ET on Slack
ETA - I hope to complete this task by EOD Tuesday, Nov 21

**Comment by allison-truhlar** [comment]
Friday progress update:

1. Progress - I haven't had time to look at this issue yet. I will look at it on Monday, and update my estimated ETA, if needed.
2. Blockers - I was working this week on another project that took up more time than expected.
3. Availability - Next week, only Mon/Tues from about noon-5pm ET each day.
4. ETA - Friday, Dec 1, EOD

**Comment by allison-truhlar** [comment]
Hi, I'm dropping this issue. I just started a new job and need to step back from HfLA for a bit. I'll message in Slack to learn more about how to step back properly!

**Comment by LRenDO** [comment]
@allison-truhlar thanks for the update! Moving this back to to be re-prioritized.

**Comment by robertnjenga** [comment]
Availability: M–F 10–2pm PT
ETA: 1/26/04  EOD
